apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -413,6 +414,107 @@ export const publishResource = async (
   return addedVersionResult
 }
 
+interface ResourceItemContentWithPath extends ResourceItemContent {
+  path: string[]
+}
+export const getBatchAncestryWithSelfQuery = async ({
```

Comment:
not super sure i understand this!

based off what i can tell, we first get all the resources inside this site and the list of resource Ids. thereafter, we find each resource and all connected resources in its path and push them to groups then return it? 

can just confirm if my understanding si correct?

Reply:
replaced with native PG code. re: https://github.com/opengovsg/isomer/pull/865#issuecomment-2628133528